pkg/xds/envoy/clusters/dns_cluster_configurer.go

```diff
@@ -0,0 +1,33 @@
+package clusters
+
+import (
+	envoy_api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+	envoy_endpoints "github.com/kumahq/kuma/pkg/xds/envoy/endpoints"
+)
+
+func DNSCluster(name string, address string, port uint32) ClusterBuilderOpt {
```

Comment:
Is that only for the Dynamic tracing, or it will allow to use DNS names in other filters too?

Reply:
Maybe in the future, but for now - not really. There are several cluster types:
1) EDS cluster - most commonly used, it relies on ClusterLoadAssignment on endpoints which cannot be DNS names.
2) Static cluster - it requires static endpoints that are IPs. It is used only for inbound cluster (which IP we know from DP definition and it should be IP because it is then used in outbounds of other DPs) and Prometheus (admin IP)
3) original dst - transparent proxy
4) strict DNS  - used here since an address can be a DNS name.